Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp

```diff
@@ -856,6 +856,22 @@ static gboolean webKitMediaSrcSendEvent(GstElement* element, GstEvent* eventTran
             source->priv->rate = rate.value();
         return forwardToAllPads ? wasEventHandledByAllStreams : wasEventHandledByAnyStream;
     }
+    case GST_EVENT_CUSTOM_DOWNSTREAM: {
+        if (gst_structure_has_name(gst_event_get_structure(event.get()), "GstEventStillFrame")) {
+            // Drain the video stream (and only that one, assuming there's only one).
+            WebKitMediaSrc* source = WEBKIT_MEDIA_SRC(element);
+            for (auto stream : source->priv->streams.values()) {
+                if (stream->track.get().type() == TrackPrivateBaseGStreamer::TrackType::Video) {
+                    GRefPtr<GstPad> sinkPeerPad = adoptGRef(gst_pad_get_peer(stream->pad.get()));
+                    stream->track->enqueueObject(adoptGRef(GST_MINI_OBJECT(event.leakRef())));
+                    return TRUE;
```

Comment:
I guess it's unlikely, but what if we have multiple video tracks? only the first one is handled.

Reply:
If we have more than one video track... we're in biiiiig trouble. That scenario hasn't ever been exercised in practical terms, even in the spec allows it in theory.

But ok, I'll change the code to continue looping and process all video tracks, just in case.